src/core/server/app/handlers/api/oEmbed/oEmbed.ts

```diff
@@ -0,0 +1,64 @@
+import { createFetch } from "coral-server/services/fetch";
+import { RequestHandler } from "coral-server/types/express";
+
+const fetchURL = (url: string, type: string) => {
+  if (type === "twitter") {
+    return `https://publish.twitter.com/oembed?url=${url}`;
+  }
+  if (type === "youtube") {
+    return `https://www.youtube.com/oembed?url=${url}`;
+  }
+
+  return "";
+};
```

Comment:
All URL parameters should be encoded with `encodeURIComponent`. While this is handled in the handler, it should instead be moved here to ensure that it's protected.

Reply:
I have done this, but had to remove the previous encoding on the url from the request. Reason being, you cannot call `encodeURIComponent` on a url twice... it will garble the url string.